16.0][MIG] delivery_postlogistics_server_env: Migration to 16.0

lehoangan opened the pull request:
Needs:

[[16.0][MIG] delivery_postlogistics: Migration to 16.0 #3]( https://github.com/lehoangan/delivery-carrier/pull/3)

pedrobaeza committed:
[ADD] delivery_auto_refresh: Auto-refresh delivery price in sales orders

Auto-refresh delivery price in sales orders
===========================================

This module automates the delivery price handling for the following cases:

* If you change any line in your draft sales order (SO), when saving, the
  delivery price will be adjusted without having to click on "→ Set price".
* If specified in the system parameter, the delivery line can be also
  auto-added when creating/saving.
* If you deliver a different quantity than the ordered one, the delivery price
  is adjusted on the linked SO when the picking is transferred.

Configuration
=============

* Activate developer mode.
* Go to *Settings > Technical > Parameters > System Parameters*.
* Locate the setting with key "delivery_auto_refresh.auto_add_delivery_line"
  or create a new one if not exists.
* Put a non Falsy value (1, True...) if you want to add automatically the
  delivery line on save.

Known issues / Roadmap
======================

* After confirming the sales order, the price of the delivery line (if exists)
  will be only updated after the picking is transferred, but not when you
  might modify the order lines.
* There can be some duplicate delivery unset/set for assuring that the refresh
  is done on all cases.
* On multiple deliveries, second and successive pickings update the delivery
  price, but you can't invoice the new delivery price.
* This is only working from user interface, as there's no way of making
  compatible the auto-refresh intercepting create/write methods from sale order
  lines.

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] delivery_auto_refresh: Make configurable also updating after transferring

cubells committed:
[MIG] : Migration to 12.0

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (2 of 2 strings)

Translation: delivery-carrier-12.0/delivery-carrier-12.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-12-0/delivery-carrier-12-0-delivery_auto_refresh/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (2 of 2 strings)

Translation: delivery-carrier-12.0/delivery-carrier-12.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-12-0/delivery-carrier-12-0-delivery_auto_refresh/zh_CN/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (2 of 2 strings)

Translation: delivery-carrier-12.0/delivery-carrier-12.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-12-0/delivery-carrier-12-0-delivery_auto_refresh/pt_BR/

pedrobaeza committed:
[FIX] delivery_auto_refresh: Avoid error when confirming sales order + add on create

As the used method now has a check for not being used when the order is confirmed, so
it collapses when you want to add the delivery line in that moment.

Previous code didn't take into account the direct creation of the sales order.

pedrobaeza committed:
[FIX] delivery_auto_refresh: Don't modify standard behavior by default

For not having bad interaction with the rest of the tests.

chienandalu committed:
[IMP] delivery_auto_refresh: keep discounts in delivery line

pedrobaeza committed:
[FIX] delivery_auto_refresh: force parameter not existing in v12

Fixes #260

pedrobaeza committed:
[FIX] delivery_auto_refresh: Create SO with default values

new() doesn't invoke defaults, so you can get an SO without company for example,
and interacts bad with other modules.

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (2 of 2 strings)

Translation: delivery-carrier-12.0/delivery-carrier-12.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-12-0/delivery-carrier-12-0-delivery_auto_refresh/sl/

sergiocorato committed:
[12.0][FIX] update delivery with multiple lines fails - add check to avoid error unmanaged

yvaucher committed:
Fix auto_refresh unit tests

In the core the onchange is now aligned with `partner_shipping_id`
instead of `partner_id`.
https://github.com/odoo/odoo/commit/b6f91e31f3c83d95b09b29d13067eb5f10e7df16

TDu committed:
[13.0][ADD] delivery_postlogistics_server_env

Adds OCA Server Environment features to Delivery Postlogistics.

The module delivery_postlogistics is in this PR:

* https://github.com/OCA/delivery-carrier/pull/323

Note removed `server-env` from `oca_dependencies.txt` temporary.

mmequignon committed:
Add delivery_postlogistics_dangerous_goods

santostelmo committed:
delivery_postlogistics: add pack_weight parameter on _prepare_attributes

phuctranfxvn committed:
[REF] delivery_carrier_label_postlogistics becomes delivery_postlogistics in 13.0

sebalix committed:
delivery_postlogistics: avoid duplication of labels

CarlosRoca13 committed:
[MIG] delivery_auto_refresh: Migration to v13.0

mmequignon committed:
Do not send disallowed chars on postlogistics api

Some chars are disallowed from postlogistics API ("|", "", "<", ">")
and output json should be sanitized

mmequignon committed:
delivery_postlogistics: Refactor tests

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-13.0/delivery-carrier-13.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-13-0/delivery-carrier-13-0-delivery_auto_refresh/

sebalix committed:
delivery_postlogistics: several fixes

* fix 'PostLogistics' terms
* make 'Package Code' required for PostLogistics delivery packaging
* show a link listing available package codes for PostLogistics delivery
  packaging
* update README

JordiMForgeFlow committed:
[13.0][IMP]delivery_auto_refresh: mark recompute_delivery_price false after refresh to prevent warning decoration

OCA-git-bot committed:
delivery_auto_refresh 13.0.1.0.1

CarlosRoca13 committed:
[FIX] delivery_auto_refresh: Error on test

When the company has different currency than the pricelist asociated to the partner. The final price is converted to the currency of the pricelist.

With this changes we achieve to avoid this recomputation and solve this problems on tests.

TT26914

OCA-git-bot committed:
delivery_auto_refresh 13.0.1.0.2

hailangvn committed:
[MOV] change module name, prepare to change file structure

mmequignon committed:
delivery_postlogistics: Handle case where no label is returned by the API

chienandalu committed:
[IMP] delivery_auto_refresh: void delivery lines on uninvoiced returns

Added new config paramerter that automates de voiding of the delivery
line for the case of returning a picking that just didn't go away from
the warehouse (due to several causes like the customer cancelling the
sale order in the last minute).

This will only be performed if the order wasn't already invoiced and the
return was set to refund.

TT30359

hailangvn committed:
[MIG] delivery_postlogistics: Migration to 14.0

The cassette `tests/fixtures/cassettes/test_store_label.yaml` should be
updated as the label was failed to decode in both 13.0 and 14.0.

* https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/ir_attachment.py#L212
* https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/ir_attachment.py#L554

We can remove the method _selection_file_type with below reasons.

* It is used in base_delivery_carrier_label but was removed on Sep 5, 2019
  https://github.com/OCA/delivery-carrier/blame/12.0/base_delivery_carrier_label/models/shipping_label.py#L15
* PR migration to 13.0 of base_delivery_carrier_label was merged on Oct 22, 2020
  https://github.com/OCA/delivery-carrier/pull/251/files
* Dependency on base_delivery_carrier_label was removed in migration to 14.0 on Apr 14 2021
  https://github.com/OCA/delivery-carrier/commit/4ba2b56e2e26ed2fd7b5b2f36b42b6f3060c45f7#diff-f16078e35000c05a9b63f5e7e04b6f5fd50eadae8b14a45aa9c1eb078031060cL12 https://github.com/OCA/delivery-carrier/commit/4ba2b56e2e26ed2fd7b5b2f36b42b6f3060c45f7#diff-08c11b571d9c0b0c509c3922c77f407ef4c429473fda9bc2d4d542520e7a4e3fR12

mmequignon committed:
[13.0] delivery_postlogistics: Add disallowed chars

joao-p-marques committed:
[IMP] *: pre-commit execution

hailangvn committed:
[MIG] delivery_postlogistics_server_env: Migration to 14.0

hailangvn committed:
[FIX] delivery_postlogistics: remove obsoleted code, get sale pickings

history_ids has been removed as below link. So the code is obsoleted.

https://github.com/odoo/odoo/commit/b3180c841101510081ee8ef9c52d205497efdd4f#diff-9a22420c4a72cc9b6d558b3d832846c2e44a4f3731c6df9579d66c6279c4804eL54-L57

_get_origin_pickings is to determine number of pickings of a sale order.
So it should get pickings from sale_id of found picking.

yankinmax committed:
[MIG] delivery_auto_refresh: Migration to 14.0

CarlosRoca13 committed:
[FIX] delivery_auto_refresh: Fix error when adding products to cart on website

The steps to reproduce the problem are these:
    - Set delivery_auto_refresh.auto_add_delivery_line.
    - Set a default delivery method on the user's partner.
    - Place a new ecommerce order with that user adding to the cart.

Doing this steps we get the error:
    Record does not exist or has been deleted.
    (Record: sale.order.line(63,), User: 1)

OCA-git-bot committed:
delivery_auto_refresh 13.0.1.0.3

hailangvn committed:
[FIX] delivery_postlogistics: avoid overriding attach_shipping_label…

… for delivery type other than postlogistics.

chienandalu committed:
[FIX] delivery_auto_refresh: behave as core

We should get as the same default carrier as Odoo does.

TT34020

OCA-git-bot committed:
delivery_auto_refresh 13.0.1.0.4

mmequignon committed:
delivery_postlogistics: Reproduce issue

mmequignon committed:
delivery_postlogistics: Fix bool has no attr split()

OCA-git-bot committed:
delivery_auto_refresh 13.0.1.0.5

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-13.0/delivery-carrier-13.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-13-0/delivery-carrier-13-0-delivery_auto_refresh/

simahawk committed:
delivery_auto_refresh: centralize get_param

simahawk committed:
delivery_auto_refresh: tests use setUpClass

chienandalu committed:
[FIX] delivery_auto_refresh: avoid singleton

If there were more than one delivery line, we'd get a singleton error
when trying to refresh due to the need of taking the discount of the
former lines.

TT35025

chienandalu committed:
[FIX] delivery_auto_refresh: don't allow to pick a not allowed carrier

When choosing the sale carrier from the header of the order form we
should filter those carriers not available for the given shipping
address the in the same way the wizard does it.

TT35200

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

OCA-git-bot committed:
delivery_postlogistics 14.0.1.0.1

ilyasProgrammer committed:
develiry_auto_refresh: enable via config params

Added parameter set_default_carrier to avoid conflicts with other modules.

OCA-git-bot committed:
delivery_auto_refresh 14.0.1.1.0

OCA-git-bot committed:
delivery_postlogistics 14.0.1.0.2

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_auto_refresh/it/

pilarvargas-tecnativa committed:
[MIG] delivery_auto_refresh: Migration to version 15.0

TT36943

Commit:
[IMP] delivery_postlogistics: black, isort, prettier

lehoangan committed:
[16.0][MIG] delivery_postlogistics: Migration to 16.0

lehoangan committed:
[IMP] delivery_postlogistics: black, isort, prettier

lehoangan committed:
[16.0][MIG] delivery_postlogistics_server_env: Migration to 16.0

dzungtran89 commented on delivery_postlogistics_server_env/README.rst:
```diff
@@ -13,15 +13,15 @@ Delivery Postlogistics Server Environment
+    :alt: OCA/delivery_postlogistics_server_env
 .. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
-    :target: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_postlogistics_server_env
+    :target: https://translation.odoo-community.org/projects/delivery_postlogistics_server_env-16-0/delivery_postlogistics_server_env-16-0-delivery_postlogistics_server_env
```
`delivery_postlogistics_server_env` should be `delivery-carrier`

lehoangan replied:
Thank you.

dzungtran89 commented:
thx, the author's email might be invalid as `JL@mail.com`, you might want to visit https://github.com/settings/emails and update your primary email to the trobz email

dzungtran89 commented:
the readme is still not correct, you might want to visit this [doc](https://github.com/OCA/maintainer-tools/blob/master/README.md#readme-generator) to see how it works

dzungtran89 reviewed (approved):
lgtm

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

mymage committed:
Translated using Weblate (Italian)

Currently translated at 10.8% (9 of 83 strings)

Translation: delivery-carrier-16.0/delivery-carrier-16.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-16-0/delivery-carrier-16-0-base_delivery_carrier_label/it/

Commit:
[UPD] Update delivery_auto_refresh.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-15.0/delivery-carrier-15.0-delivery_auto_refresh
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-15-0/delivery-carrier-15-0-delivery_auto_refresh/

rousseldenis committed:
[FIX] stock_picking_delivery_link: Add view for picking type option

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 40.9% (34 of 83 strings)

Translation: delivery-carrier-16.0/delivery-carrier-16.0-base_delivery_carrier_label
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-16-0/delivery-carrier-16-0-base_delivery_carrier_label/sl/

OCA-git-bot committed:
Merge PR #619 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update delivery_postlogistics.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

rousseldenis committed:
[16.0][ADD] delivery_automatic_package

pedrobaeza committed:
[FIX] delivery_auto_refresh: Don't fail on fixed rate carriers

Steps to reproduce:

- Create a delivery method (delivery.carrier) with a fixed rate.
- Enable the system parameter "delivery_auto_refresh.refresh_after_picking".
- Create a sales order with such delivery method and storable products.
- Confirm the sales order.
- Validate the picking.

Result: It can't be validated with the message "There is no matching
delivery rule.".

Explanation: Previous code computes the shipping values and call
directly to `_get_price_from_picking`, which uses rules and doesn't
take into account the rest of possibilities (integration or fixed
price).

We keep such computation for carriers based on rules, as they are not
properly updated, but added the possibility of writing back the price
for other carriers with fixed price or integrations, using the
`carrier_price` field that contains it.

TT43533

OCA-git-bot committed:
Merge PR #624 into 16.0

Signed-off-by rousseldenis

Commit:
[UPD] Update delivery_state.pot

OCA-git-bot committed:
stock_picking_delivery_link 16.0.1.1.0

OCA-git-bot committed:
[UPD] addons table in README.md

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-16.0/delivery-carrier-16.0-delivery_state
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-16-0/delivery-carrier-16-0-delivery_state/

OCA-git-bot committed:
delivery_auto_refresh 15.0.1.0.1

RabbitJon-S73 committed:
[MIG] delivery_auto_refresh: migration to 16.0

OCA-git-bot committed:
Merge PR #651 into 16.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update delivery_auto_refresh.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
Merge PR #642 into 16.0

Signed-off-by bguillot

Commit:
[UPD] Update delivery_automatic_package.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py
